### PR TITLE
fix(taskframework): running task log is not found when close task-framework &&  send mistake alarm

### PR DESCRIPTION
**Pull request by krihy**
#### What this PR does / why we need it:

1. try to stop task before destroy executor, less task do final some thing, eg release db connection and upload log when close task-framework
image
2. add `ODC_EXECUTOR_CANNOT_BE_DESTROYED` avoid send mistake `TASK_EXECUTOR_DESTROY_FAILED` alarm 
3. upload task result with row lock avoid cover latest value
4. format arguments to string in `JobException` 

#### Which issue(s) this PR fixes:

test passed

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java [hunk 1]
```diff
@@ -314,16 +314,18 @@ public void handleResult(TaskResult taskResult) {
...
-        updateJobScheduleEntity(taskResult);
-        taskResultPublisherExecutor
-                .execute(() -> publisher.publishEvent(new DefaultJobProcessUpdateEvent(taskResult)));
-        if (publisher != null && taskResult.getStatus() != null && taskResult.getStatus().isTerminated()) {
-            taskResultPublisherExecutor.execute(() -> publisher
-                    .publishEvent(new JobTerminateEvent(taskResult.getJobIdentity(), taskResult.getStatus())));
-            if (taskResult.getStatus() == JobStatus.FAILED) {
-                AlarmUtils.alarm(AlarmEventNames.TASK_EXECUTION_FAILED,
-                        MessageFormat.format("Job execution failed, jobId={0}", taskResult.getJobIdentity().getId()));
+        int rows = updateJobScheduleEntity(taskResult);
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java:317]: add status condition for updateJobScheduleEntity, use nativeQuery

**Review by yizhouxw — COMMENTED** [review]
please change PR title sytle, use scenario description, not solution view

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java (hunk 1 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java]: yes, I use nativeQuery `updateReportResult` with id and old status to update result

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by smallsheeeep — APPROVED** [review]
LGTM (Better to simplify the title of this PR)

**Comment by krihy** [comment]
> LGTM (Better to simplify the title of this PR)

DONE

**Pull request merged**